### PR TITLE
Allows wearing plastic bags on head

### PR DESCRIPTION
**Pull request by unid15**
Wearing a plastic bag prevents you from breathing. It also hides your face!

You can't wear bags which contain something, and you can't put anything inside worn bags

### CODE REVIEW

File: code/game/objects/items/weapons/storage/bags.dm [hunk 1]
```diff
@@ -66,9 +66,29 @@
...
+
+	max_combined_w_class = 0 //Can't put anything in once equipped!
+
+/obj/item/weapon/storage/bag/plasticbag/u_equip()
+	.=..()
+
+	max_combined_w_class = initial(max_combined_w_class) //Can put stuff in once unequipped
+
+/obj/item/weapon/storage/bag/plasticbag/suicide_act(mob/user)
+	viewers(user) << "<span class='danger'>[user] puts the [src.name] over \his head and tightens the handles around \his neck! It looks like \he's trying to commit suicide.</span>"
```
Comment (PJB3005) [inline code/game/objects/items/weapons/storage/bags.dm]: Should probably be `visible_message()`

**Comment by wwjnc** [comment]
potential harm
potential harm
potential harm
woop woop bizwarn

**Comment by Intigracy** [comment]
code/game/objects/items/weapons/storage/bags.dm:84:error: u_equip: undefined proc

File: code/game/objects/items/weapons/storage/bags.dm [hunk 2]
```diff
@@ -66,9 +66,26 @@
 	can_hold = list() // any
 	cant_hold = list("/obj/item/weapon/disk/nuclear")
 
-	suicide_act(mob/user)
-		viewers(user) << "<span class='danger'>[user] puts the [src.name] over \his head and tightens the handles around \his neck! It looks like \he's trying to commit suicide.</span>"
-		return(OXYLOSS)
+	slot_flags = SLOT_BELT | SLOT_HEAD
+	flags = FPRINT | BLOCK_BREATHING
```
Comment (Intigracy) [inline code/game/objects/items/weapons/storage/bags.dm]: BLOCK_GAS_SMOKE_EFFECT ?
I'm assuming if you couldn't breath that you wouldn't be able to inhale smoke.

**Comment by Arthurdentist** [comment]
if you are wearing a breathing mask and have your oxygen turned on, will this still block you from breathing anyway?

**Comment by Unfit** [comment]
@Arthurdentist, it shouldn't

**Comment by Shadowmech88** [comment]
This is essentially a "hold breath" function.
Which is nice, because sometimes I'll take the suffocation damage in order to not breathe plasma, N2O, fire, vacuum, or what have you.

**Comment by dylanstrategie** [comment]
It might be better down the line to add an option to hold your breath. A verb should do

More often than not, not breathing is the most logical solution to an atmospherics problem, outside of pure vacuum where it is extremely dangerous

**Comment by N3X15** [comment]
Conflicts

File: code/modules/mob/living/carbon/human/life/handle_breath.dm [hunk 3]
```diff
@@ -37,7 +37,11 @@
 
 		//No breath from internal atmosphere so get breath from location
 		if(!breath)
-			if(isobj(loc))
+			if(head && (head.flags & BLOCK_BREATHING)) //Worn items which block breathing are handled first
+				//
+			else if(wear_mask && (wear_mask.flags & BLOCK_BREATHING))
+				//
```
Comment (N3X15) [inline code/modules/mob/living/carbon/human/life/handle_breath.dm]: Should be 
```
var/can_breathe = 1
if(head && (head...))
    can_breathe = 0
if(wear_mask && ...))
    can_breathe = 0
if(can_breathe && isobj(loc))
    ...
else if(can_breathe && isturf(loc))
```

**Pull request merged**